homework/App.js

```diff
@@ -16,14 +16,78 @@ class App {
     // 1. Create the fixed HTML elements of your page
     // 2. Make an initial XMLHttpRequest using Util.fetchJSON() to populate your <select> element
 
+    const BodyEl = document.body;
+
     const root = document.getElementById('root');
 
-    Util.createAndAppend('h1', root, { text: 'It works!' }); // TODO: replace with your own code
+    // Create div element 'select' in  document body to hold the label element and list box.
+
+    Util.createAndAppend('div', BodyEl, { id: 'select-container' });
+    const select = document.getElementById('select-container');
+    Util.createAndAppend('LABEL', select, {
+      text: 'HYF Repositories: ',
+      id: 'label',
+      for: 'repo',
+    });
+    Util.createAndAppend('select', select, { id: 'select-repo' });
+
+    // Create two div elements section1 and section2 under 'Root' div to have
+    // section1 - Repository Information.
+    // section2 - Contributions.
+
+    Util.createAndAppend('div', root, { id: 'repo-details' });
+    Util.createAndAppend('div', root, { id: 'contributor-information' });
+
+    // Insert section1 before section2 div element under 'root' div.
+    const newNode = document.getElementById('contributor-information');
+    const referenceNode = document.querySelector('repo-details');
+    root.insertBefore(newNode, referenceNode);
+
+    // Insert Select div first in the body before root div.
+    BodyEl.insertBefore(select, document.getElementById('root'));
 
     try {
       const repos = await Util.fetchJSON(url);
+
       this.repos = repos.map(repo => new Repository(repo));
-      // TODO: add your own code here
+
+      const sortRepoName = [];
+      // push all the HYP repo names to sort array.
+
+      Object.keys(repos).forEach(key => {
+        sortRepoName.push(repos[key].name);
+      });
+      // sort the repo name using sort function and localeComapare for uppercase and lowercase sorting.
+      sortRepoName.sort((a, b) => a.localeCompare(b));
+
+      const selectBox = document.getElementById('select-repo');
+
+      // Create Option under Select element and attach the  same with SELECT element.
+
+      sortRepoName.forEach(loadrepo => {
+        const option = document.createElement('option');
+        option.value = loadrepo;
+        option.text = loadrepo;
+        selectBox.appendChild(option);
+      });
+
+      let keyIndex;
+      console.log(selectBox.value);
+      Object.keys(repos).forEach(key => {
+        if (selectBox.value === repos[key].name) {
+          keyIndex = key;
+        }
+      });
+      this.fetchContributorsAndRender(keyIndex);
+
+      selectBox.addEventListener('change', () => {
+        Object.keys(repos).forEach(key => {
+          if (selectBox.value === repos[key].name) {
+            keyIndex = key;
+          }
+        });
+        this.fetchContributorsAndRender(keyIndex);
+      });
     } catch (error) {
```

Comment:
This catch could probably just wrap the `const repos = await Util.fetchJSON(url);` line. Being picky though.